akka-persistence-typed/src/main/scala/akka/persistence/typed/PersistenceId.scala

```diff
@@ -131,13 +131,23 @@ object PersistenceId {
  * Unique identifier in the backend data store (journal and snapshot store) of the
  * persistent actor.
  */
-final class PersistenceId private (val id: String) {
-  if (id eq null)
+final class PersistenceId private (
+    val entityId: String,
+    maybeHint: Option[String] = None,
+    separator: String = PersistenceId.DefaultSeparator) {
```

Comment:
I need to think some about this. In the end it's just a String in the journal so for example in read-side (consuming eventsByTag) we can't reliably re-create a `PersistenceId` with the same fields as the original from that String.

Reply:
Ah, I see.

Actually, your comment made me realize this PR is wrong. 

The current implementation could return invalid values for `persistenceId.entityId` when the instance is built using `PersistenceId.ofUniqueId("hint|123")`. When the user only has the `id` but doesn't have a reverse function to destruct it into `entityId`, `entityTypeHint` and `separator` then `persistenceId.entityId` should return `None` (or some form of failure).

Reply:
exactly, that was what I was concerned about